src/Components/Reports/SystemsPDFReport.js

```diff
@@ -1,51 +1,72 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { DownloadButton } from '@redhat-cloud-services/frontend-components-pdf-generator';
-import { PDF_REPORT_PER_PAGE, SYSTEMS_PDF_REPORT_TITLE } from '../../constants';
+import { PDF_REPORT_PER_PAGE, REPORT_NOTIFICATIONS, SYSTEMS_PDF_REPORT_TITLE } from '../../constants';
 import { fetchSystems } from '../../Utilities/api';
 import { formatData, generateFilterText, getSystemsReportFileName } from './Util';
 import propTypes from 'prop-types';
 import { SystemsTablePage } from './Common/SystemsTablePage';
 import { SystemsFirstPage } from './Common/SystemsFirstPage';
+import { useNotification } from './Common/useNotification';
 
-const generateSystemsPDFReport = async (filters, orderBy, orderHow) => {
+export const DownloadSystemsPDFReport = ({ filters, orderBy, orderHow, ...props }) => {
+    const reportFileName = getSystemsReportFileName();
+    const [addNotification, clearNotifications] = useNotification();
+    const { start, success, failure } = REPORT_NOTIFICATIONS;
 
-    const { systemsReportFirstPage, systemsReportRestPages } = PDF_REPORT_PER_PAGE;
+    useEffect(() => {
+        addNotification(start);
+    }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-    // Table rows
-    const fetchSystemParams = {
-        filters,
-        stateFilter: filters.stateFilter,
-        orderBy,
-        orderHow
-    };
-    const systemsResponse = await fetchSystems(fetchSystemParams);
-    const pdfData = formatData(systemsResponse.data, 'pdf');
+    const generateSystemsPDFReport = async (filters, orderBy, orderHow) => {
 
-    // first page description and data
-    const totalSystems = systemsResponse?.meta?.count;
-    const filterText = generateFilterText(filters);
-    const firstPageData =  pdfData.splice(0, systemsReportFirstPage);
+        const { systemsReportFirstPage, systemsReportRestPages } = PDF_REPORT_PER_PAGE;
```

Comment:
```suggestion
        const { firstPageCount, otherPageCount } = PDF_RECORDS_PER_PAGE;
```
Could you rename these keys something like ^^ ?

Reply:
Thanks for the suggestions @kgaikwad 🙌🏻 
I have updated the same and will push the changes in the latest commit.